docarray/array/storage/weaviate/getsetdel.py

```diff
@@ -0,0 +1,206 @@
+import itertools
+from collections.abc import Iterable as _Iterable
+from typing import (
+    Sequence,
+    Iterable,
+    Any,
+)
+
+from ..base.getsetdel import BaseGetSetDelMixin
+from .... import Document
+
+
+class GetSetDelMixin(BaseGetSetDelMixin):
+    """Provide concrete implmentation for ``__getitem__``, ``__setitem__``,
+    and ``__delitem__`` for ``DoucmentArrayWeaviate``"""
+
+    def _getitem(self, wid: str) -> 'Document':
+        """Helper method for getting item with weaviate as storage
+
+        :param wid: weaviate id
+        :raises KeyError: raise error when weaviate id does not exist in storage
+        :return: Document
+        """
+        resp = self._client.data_object.get_by_id(wid, with_vector=True)
+        if not resp:
+            raise KeyError(wid)
+        return Document.from_base64(resp['properties']['_serialized'])
+
+    def _setitem(self, wid: str, value: Document):
+        """Helper method for setting an item with weaviate as storage
+
+        :param wid: weaviate id
+        :param value: the new document to update to
+        """
+        if self._client.data_object.exists(wid):
+            self._client.data_object.delete(wid)
```

Comment:
I would suggest moving `delete` ops after the `self._doc2weaviate_create_payload`.

Reply:
Addressed, thanks.